src/utils.ts

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+module jupyter.utils {
+
+/*
+ * Copy the contents of one object to another, recursively.
+ *
+ * http://stackoverflow.com/questions/12317003/something-like-jquery-extend-but-standalone
+ */
+export
+function extend(target: any, source: any): any {
+  target = target || {};
+  for (var prop in source) {
+    if (typeof source[prop] === 'object') {
+      target[prop] = extend(target[prop], source[prop]);
+    } else {
+      target[prop] = source[prop];
+    }
+  }
+  return target;
+}
+
+
+/*
+ * Get a uuid as a string.
+ *
+ * http://www.ietf.org/rfc/rfc4122.txt
+ */
+export
+function uuid(): string {
+  var s: string[] = [];
+  var hexDigits = "0123456789ABCDEF";
+  for (var i = 0; i < 32; i++) {
+    s[i] = hexDigits.substr(Math.floor(Math.random() * 0x10), 1);
```

Comment:
this could be `hexDigits.charAt` instead of `substr`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charAt